sdk/tables/Azure.Data.Tables/src/Extensions/DictionaryTableExtensions.cs

```diff
@@ -104,7 +104,7 @@ internal static void CastAndRemoveAnnotations(this IDictionary<string, object> e
                 entity[annotation] = typeAnnotationsWithKeys[annotation].typeAnnotation switch
                 {
                     TableConstants.Odata.EdmBinary => Convert.FromBase64String(entity[annotation] as string),
-                    TableConstants.Odata.EdmDateTime => DateTime.Parse(entity[annotation] as string, CultureInfo.InvariantCulture, DateTimeStyles.RoundtripKind),
+                    TableConstants.Odata.EdmDateTime => DateTimeOffset.Parse(entity[annotation] as string, CultureInfo.InvariantCulture, DateTimeStyles.RoundtripKind),
```

Comment:
Nit - is this visible to users?  Would it need a Changelog entry?

Reply:
We were ultimately casting to `DateTimeOffset` in `TableEntity` so this is just an internal implementation detail and a bug fix. Previously, the Timestamp property always returned null.